src_blog/_posts/New-Release-Grain-v0.4.0-Cereal.md

```diff
@@ -0,0 +1,67 @@
+
+We have preliminary support for NEAR in this release, but things will get **really** interesting once we write the NEAR SDK for Grain. This is some of our most pressing work now that the release is out—check back soon to start writing smart contracts in Grain. 📈
+
+## Tools
+
+Another important goal for this release was to improve the tooling experience for our end users and for ourselves. Towards this goal, we added two new commands to the Grain CLI: `grain format` and `grain doc`.
+
+### Grain format
+
+Grain format provides an opinionated code formatter tool for Grain source code. This tool was an absolutely herculean effort by Marcus, the core team member we introduced in our last release post and maintainer of the Grain Language Server Protocol.
+
+I won’t go in-depth on the formatter in this post, but Marcus wrote up an excellent [blog post](https://grain-lang.org/blog/2021/09/04/grain-formatter/) about it! I recommend you give it a read, and grab our latest vscode extension to have format-on-save; I’ve been using it for a few days and it is a great experience.
+
+### Grain doc
+
+Additionally, I finally finished the initial work around our documentation tool. This provides a JSDoc-like experience for your Grain code. We’ve already begun using it to document the standard library, so we’ll be able to keep our API docs consistently updated.
+
+You can check out the features and syntax in my preview [blog post](https://grain-lang.org/blog/2021/08/27/tooling-preview-graindoc/). I’m particularly happy that the tool infers the types of your parameters and return values, so you don’t have to worry about keeping them in-sync.
+
+## Callee-owned values
+
+This is a bit of an internal change, but it can cause a lot of ripples throughout the ecosystem. We changed the way our compiler outputs Garbage Collector code so that values passed to a function are owned and cleaned up by the function being called. Previously, the caller would `decRef` the values after a function returned; however, we found that this caused some memory leaks in certain scenarios. By changing this memory ownership convention, we now know all parameters are `decRef`ed by the time the function returns.
+
+This only has ramifications to you if you were using the `@disableGC` annotation on your Grain code. While this attribute is not recommended for end-user code, we use it in the stdlib and some library authors might need to use it. If you are in need of this, please get in touch with us to understand the new calling convention.
```

Comment:
I think we should go into a lot less detail here. It's worth calling out, but I don't think we should really get into it in this post. I think a technical post about it (hi @peblair) would be good.
```suggestion
This is a bit of an internal change. We changed Grain's garbage collection calling convention such that a function that is called owns the references to its arguments rather than the caller. This was done to help us assure that all allocated memory is properly reclaimed.

This shouldn't at all affect how you use Grain! This only has ramifications to you if you were using Grain's `unsafe` libraries for low-level code or bindings to WebAssembly host functions. This is typically only done by some library authors (and isn't recommended otherwise). If you are actively using the `unsafe` libraries, please get in touch with us to understand the new calling convention.
```